pkg/operator/controller/ingress/status.go

```diff
@@ -136,11 +136,15 @@ func MergeConditions(conditions []operatorv1.OperatorCondition, updates ...opera
 		for j, cond := range conditions {
 			if cond.Type == update.Type {
 				add = false
-				if conditionChanged(cond, update) {
-					conditions[j].Status = update.Status
-					conditions[j].Reason = update.Reason
-					conditions[j].Message = update.Message
-					conditions[j].LastTransitionTime = now
+				if statusChange, otherChange := controller.ConditionChanged(conditions[j].Message, update.Message, string(conditions[j].Status), string(update.Status), conditions[j].Reason, update.Reason); statusChange || otherChange {
+					if otherChange {
+						conditions[j].Reason = update.Reason
+						conditions[j].Message = update.Message
+					}
+					if statusChange {
+						conditions[j].Status = update.Status
+						conditions[j].LastTransitionTime = now
+					}
```

Comment:
We should add a test case to verify that the new logic correctly handles the issue in the Jira ticket.  What do you think about adding the following to `Test_MergeConditions`?  (Sorry, it ends up needing being rather verbose in order to capture the critical details.)
```go
		"update message": {
			conditions: []operatorv1.OperatorCondition{{
				Type:               "CanaryChecksSucceeding",
				Status:             "False",
				LastTransitionTime: metav1.NewTime(start),
				Reason:             "CanaryChecksRepetitiveFailures",
				Message: "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x5 over 4m40s)",
			}},
			updates: []operatorv1.OperatorCondition{{
				Type:               "CanaryChecksSucceeding",
				Status:             "False",
				LastTransitionTime: metav1.NewTime(start),
				Reason:             "CanaryChecksRepetitiveFailures",
				Message: "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x14 over 15m10s)",
			}},
			expected: []operatorv1.OperatorCondition{{
				Type:               "CanaryChecksSucceeding",
				Status:             "False",
				// Keep the old lastTransitionTime.
				LastTransitionTime: metav1.NewTime(start),
				Reason:             "CanaryChecksRepetitiveFailures",
				// Use the updated message.
				Message: "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x14 over 15m10s)",
			}},
		},
		"update message and status": {
			conditions: []operatorv1.OperatorCondition{{
				Type:               "CanaryChecksSucceeding",
				Status:             "True",
				LastTransitionTime: metav1.NewTime(start),
				Reason:             "CanaryChecksSucceeding",
				Message: "Canary route checks for the default ingress controller are successful",
			}},
			updates: []operatorv1.OperatorCondition{{
				Type:               "CanaryChecksSucceeding",
				Status:             "False",
				LastTransitionTime: metav1.NewTime(middle),
				Reason:             "CanaryChecksRepetitiveFailures",
				Message: "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x5 over 4m40s)",
			}},
			expected: []operatorv1.OperatorCondition{{
				Type:               "CanaryChecksSucceeding",
				Status:             "False",
				// Update lastTransitionTime if status changed.
				LastTransitionTime: metav1.NewTime(later),
				Reason:             "CanaryChecksRepetitiveFailures",
				Message: "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x5 over 4m40s)",
			}},
		},
```

Reply:
Thanks.  Made this change and added a couple of other tests.